ImperatorToCK3/Source/Mappers/LocalizationMapper/LocalizationMapper.cpp

```diff
@@ -0,0 +1,121 @@
+			scrapeLanguage("english", mod.second + "/localization");
+			scrapeLanguage("french", mod.second + "/localization");
+			scrapeLanguage("german", mod.second + "/localization");
+			scrapeLanguage("russian", mod.second + "/localization");
+			scrapeLanguage("spanish", mod.second + "/localization");
+		}
+	}
+
+	LOG(LogLevel::Info) << ">> " << localizations.size() << " words read.";
+}
+
+void mappers::LocalizationMapper::scrapeLanguage(const std::string& language, const std::string& path)
+{
+	if (!Utils::DoesFolderExist(path + "/" + language))
+		return;
+	auto filenames = Utils::GetAllFilesInFolderRecursive(path + "/" + language);
+	for (const auto& file : filenames)
+	{
+		std::ifstream fileStream(path + "/" + language + "/" + file);
+		if (fileStream.is_open())
+			scrapeStream(fileStream, language);
+		fileStream.close();
+	}
+}
+
+void mappers::LocalizationMapper::scrapeStream(std::istream& theStream, const std::string& language)
+{
+	while (!theStream.eof())
+	{
+		std::string line;
+		getline(theStream, line);
+
+		if (line[0] == '#' || line.length() < 4)
+			continue;
+
+		const auto sepLoc = line.find_first_of(':');
+		if (sepLoc == std::string::npos)
+			continue;
+		const auto key = line.substr(1, sepLoc-1);
+		const auto newLine = line.substr(sepLoc + 1, line.length());
+		const auto quoteLoc = newLine.find_first_of('\"');
+		const auto quote2Loc = newLine.find_last_of('\"');
+		if (quoteLoc == std::string::npos || quote2Loc == std::string::npos || quote2Loc - quoteLoc == 0)
+			continue;
+		const auto value = newLine.substr(quoteLoc + 1, quote2Loc - quoteLoc - 1);
+
+		if (localizations.count(key))
+		{
+			if (language == "english")
+				localizations[key].english = value;
+			if (language == "french")
+				localizations[key].french = value;
+			if (language == "german")
+				localizations[key].german = value;
+			if (language == "russian")
+				localizations[key].russian = value;
+			if (language == "spanish")
+				localizations[key].spanish = value;
+		}
+		else
+		{
+			LocBlock newBlock;
+			if (language == "english")
+				newBlock.english = value;
+			if (language == "french")
+				newBlock.french = value;
+			if (language == "german")
+				newBlock.german = value;
+			if (language == "russian")
+				newBlock.russian = value;
+			if (language == "spanish")
+				newBlock.spanish = value;
+			localizations.insert(std::pair(key, newBlock));
+		}
+	}
+}
+
+std::optional<mappers::LocBlock> mappers::LocalizationMapper::getLocBlockForKey(const std::string& key) const
+{
+	const auto& keyItr = localizations.find(key);
+	if (keyItr == localizations.end())
+		return std::nullopt;
+
+	if (!keyItr->second.english.empty() && (keyItr->second.spanish.empty() || keyItr->second.russian.empty() || keyItr->second.german.empty() || keyItr->second.french.empty()))
+	{
+		auto newBlock = keyItr->second;
+		if (newBlock.spanish.empty())
+			newBlock.spanish = newBlock.english;
+		if (newBlock.russian.empty())
```

Comment:
The game will have a merry mix of english in latin and russian in cyrilic. :)

Reply:
That's still better than seeing the unlocalized code names.